tests/ui/consts/const-block-const-bound.stderr

```diff
@@ -1,44 +1,30 @@
 error[E0277]: can't drop `UnconstDrop` in const contexts
-  --> $DIR/const-block-const-bound.rs:20:11
+  --> $DIR/const-block-const-bound.rs:20:9
    |
 LL |         f(UnconstDrop);
-   |         - ^^^^^^^^^^^ the trait `~const Destruct` is not implemented for `UnconstDrop`
-   |         |
-   |         required by a bound introduced by this call
+   |         ^^^^^^^^^^^^^^ the trait `~const Destruct` is not implemented for `UnconstDrop`
    |
    = note: the trait bound `UnconstDrop: ~const Destruct` is not satisfied
-note: required by a bound in `f`
-  --> $DIR/const-block-const-bound.rs:6:15
-   |
-LL | const fn f<T: ~const Destruct>(x: T) {}
-   |               ^^^^^^^^^^^^^^^ required by this bound in `f`
 help: consider borrowing here
    |
-LL |         f(&UnconstDrop);
-   |           +
-LL |         f(&mut UnconstDrop);
-   |           ++++
+LL |         &f(UnconstDrop);
+   |         +
+LL |         &mut f(UnconstDrop);
+   |         ++++
```

Comment:
This is a regression that has something to do with the spans and how trait selection is different in typeck and constck. I will look into this.

Reply:
It looks like the span of the `ObligationCause` is what caused this. Kind of hard to fix since we don't have access to the machinery that typeck can use.